spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioServerConnectionFactory.java

```diff
@@ -123,6 +124,7 @@ public void run() {
 			if (logger.isInfoEnabled()) {
 				logger.info(this + " Listening");
 			}
+			publishServerListeningEvent(getPort());
```

Comment:
It isn't clear from here on GH, but you emit this event for Nio before `setListening(true);`.
Looks like inconsistency with Net and doesn't reflect the documentation below.

Although the moving it to there doesn't help me to overcome failures:
```
ConnectionFactoryTests. testObtainConnectionIdsNet
TcpNioConnectionTests. int3453RaceTest
```